sycl/doc/extensions/proposed/sycl_ext_intel_grf_size.asciidoc

```diff
@@ -0,0 +1,143 @@
+  sycl_ext_oneapi_kernel_properties]
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices and
+only when using the Level Zero backend or OpenCL backend.
+Attempting to use this extension in kernels that run on other devices or
+backends may result in undefined behavior.  Be aware that the compiler
+is not able to issue a diagnostic to warn you if this happens.
+
+== Overview
+
+There are devices where the size of the general register file (GRF) used by a kernel is 
+configurable. Developers may want to change the GRF size based on their
+application. This extension adds the kernel property `grf_size` which provides a way
+to specify the GRF size.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_GRF_SIZE` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== GRF Size Property
+
+The `grf_size` kernel property provides a way to specify the GRF size used by a kernel. 
+It accepts a single integer value. It is defined as follows:
+```c++
+namespace sycl::ext::intel::experimental {
+
+struct grf_size_key {
+  template <int Size>
+  using value_t = 
+      oneapi::experimental::property_value<grf_size_key, 
+                                           std::integral_constant<int, Size>>;
+};
+
+template <int Size>
+inline constexpr grf_size_key::value_t<Size> grf_size;
+
+} // namespace sycl::ext::intel::experimental
+```
+The supported values are as follows:
+[%header,cols="1,5"]
+|===
+|GPU |Supported Values
+| PVC | 128 (small register file), 256 (large register file)
+| DG2 | 128 (small register file), 256 (large register file)
+|===
+
+Providing a value not consistent with the supported values may result in undefined behavior.
+
+=== Using the property in a kernel
+
+A simple example of using this extension is below.
+
+```c++
+namespace syclex = sycl::ext::oneapi::experimental;
+namespace intelex = sycl::ext::intel::experimental;
+{
+  ...
+  syclex::properties kernel_properties{intelex::grf_size<256>};
```

Comment:
```suggestion
namespace onepiex = sycl::ext::oneapi::experimental;
namespace intelex = sycl::ext::intel::experimental;
{
  ...
  onepiex::properties kernel_properties{intelex::grf_size<256>};
```
I think `syclex` stands for SYCL extension. It's reasonable name when we either alias this name to `sycl::ext` namespace or if there are other alias to `sycl::ext` nested namespaces. In this case, it's better to avoid using `syclex` name to avoid ambiguity.

Reply:
> `namespace syclex = sycl::ext::oneapi;  /* or sycl::ext::oneapi::experimental */`

This `or` might be confusing. If we want to use the same name across all the specs we can't assign `sycl::ext::oneapi::experimental`. We have to use `namespace syclex = sycl::ext::oneapi;` definition.

If namespaces other than `sycl::ext::oneapi` are rarely used in the specs, maybe we can avoid using aliases for them and use full declarations?

> I certainly don't mind changing if that's the consensus.

How do we decide if we have a consensus? :)
I think using `syclex` for `sycl::ext::oneapi` impairs readability (using it for `sycl::ext` is more logical), but it's a minor issue. The code snippets in the specs are quite short and I think aliases are defined in each of them. I'm okay if we leave it as is.